pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -168,6 +170,40 @@ impl<T: Config> Pallet<T> {
 		Ok(().into())
 	}
 
+	/// Schedules a [DelegationAction::Leave] for the delegator, towards a given collator.
+	pub(crate) fn delegation_schedule_leave(
+		collator: T::AccountId,
+		delegator: T::AccountId,
+		bonded_amount: BalanceOf<T>,
+	) -> DispatchResultWithPostInfo {
+		let mut state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		let mut scheduled_requests = <DelegationScheduledRequests<T>>::get(&collator);
+
+		// remove any other scheduled request, if not [DelegationAction::Leave]
+		if let Some(index) = scheduled_requests
+			.iter()
+			.position(|req| req.delegator == delegator)
```

Comment:
It's `O(n)` where `n` is the number of delegators, that can be a problem

Reply:
This is bounded by max delegations per candidate (`350` on moonriver), so not a huge issue. If there's a concern then we should change this in other functions as well.

Reply:
It's not all delegation scheduled requests, only the ones per the input collator. So it is max 350.

Reply:
Oooh, I see, in this case it's fine

Reply:
We should benchmark this against the size of `scheduled_request` to see if it's significant. Hopefully it's not -- it's just iterating and 350 isn't that much.

ideas for improving this, if needed:
* sort `DelegationScheduledRequests` (if not already sorted)
* add a weight hint for the number of requests needing to be iterated
* charge for the max (e.g. 350 iterations) but refund